make_image.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+
+set -o errtrace
+set -o nounset
+
+ROOT="$( cd $( dirname $0 ); pwd )"
+
+TOOLCHAIN_DIR_DEFAULT="/opt/grisp/grisp2-rtems-toolchain/rtems/5"
+BAREBOX_IMG_FILENAME="barebox-phytec-phycore-imx6ul-emmc-512mb.img"
+BAREBOX_RELDIR="../../barebox"
+OUTPUT_FILE="$PWD/grisp2_emmc.img.gz"
+LOOPBACK_ID="99"
+
+TOOLCHAIN_DIR=${GRISP_TOOLCHAIN:-$TOOLCHAIN_DIR_DEFAULT}
+BAREBOX_IMG=""
+ERLANG_APP_DIR="$ROOT"
+ERLANG_APP_NAME="grisp_demo"
+ERLANG_APP_VSN="0.1.0"
+FORCE=0
+TRUNCATE=1
+
+function usage() {
+	local code="${1:-0}"
+	echo "USAGE: $0 [-h] [-f] [-n] [-t TOOLCHAIN_ROOT] [-i BOOTLOADER_IMG] [-a ERLANG_APP] [-n APP_NAME] [-v APP_VSN] [-l LOOPBACK_ID] [-o OUTPUT_IMG]"
+	echo "  -h: Show this help"
+	echo "  -f: Force the overwrite of the output files"
+	echo "  -n: Do not truncate the second partition of the image"
+	echo "  -t TOOLCHAIN_ROOT: GRiSP 2 toolchain directory"
+	echo "    default: $TOOLCHAIN_DIR_DEFAULT"
+	echo "  -i BOOTLOADER_IMG: GRiSP 2 bootloader image"
+	echo "    default: TOOLCHAIN_ROOT/$BAREBOX_RELDIR/$BAREBOX_IMG_FILENAME"
+	echo "  -a ERLANG_APP: Erlang application to be deployed on partition A"
+	echo "    default: $ERLANG_APP_DIR"
+	echo "  -n APP_NAME: Name of the erlang release do deploy"
+	echo "    default: $ERLANG_APP_NAME"
+	echo "  -v APP_VSN: Version of the erlang release do deploy"
+	echo "    default: $ERLANG_APP_VSN"
+	echo "  -l LOOPBACK_ID: Number of the loopback device to use"
+	echo "    default: $LOOPBACK_ID"
+	echo "  -0 OUTPUT_IMG: Output image"
+	echo "    default: $OUTPUT_FILE"
+	exit $code
+}
+
+function error() {
+	local code="$1"
+	shift
+	local msg="$@"
+	echo "ERROR: $msg ($code)"
+	usage
+}
+
+while getopts "hft:i:a:l:o:" o; do
+    case "${o}" in
+    	h)
+			usage
+			;;
+		f)
+			FORCE=1
+			;;
+		n)
+			TRUNCATE=0
+			;;
+        t)
+            TOOLCHAIN_DIR=${OPTARG}
+            ;;
+        i)
+			BAREBOX_IMG=${OPTARG}
+			;;
+		a)
+			ERLANG_APP_DIR=${OPTARG}
+			;;
+		n)
+			ERLANG_APP_NAME=${OPTARG}
+			;;
+		v)
+			ERLANG_APP_VSN=${OPTARG}
+			;;
+		l)
+			LOOPBACK_ID=${OPTARG}
+			;;
+		o)
+			OUTPUT_FILE=${OPTARG}
+			;;
+        *)
+            usage 1
+            ;;
+    esac
```

Comment:
The indentation here looks really off. Is it intentional? There seems to be mixed indentation with both tabs and spaces in all files. Please replace with only spaces or only tabs.